app/controllers/materials_controller.rb

```diff
@@ -13,6 +13,19 @@ def show
     @material = Material.find(params[:id])
   end
 
+  def compare
+    num_materials = params[:num_to_compare].to_i || 0
+    if params[:commit] == "Add material"
+      num_materials += 1
+    end
+
+    @materials = []
+    (0..num_materials-1).each do |i|
+      uuid =  params["material_#{i}_uuid"]
+      @materials.append(Material.exists?(uuid) ? Material.find(uuid) : nil)
+    end
+  end
```

Comment:
Be good to move the logic here in the controller into a service object that takes the params as input and performs the manipulation to get the necessary data.